src/mlpack/core/cv/metrics/ssim_index.hpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * @file ssim_index.hpp
+ * @author Utkarsh Rai
+ *
+ * The SSIM (Structural Similarity) metric.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_METRICS_SSIM_HPP
+#define MLPACK_CORE_CV_METRICS_SSIM_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+	namespace cv {
+
+		class SSIMIndex {
+			public:
```

Comment:
Hey @UtR491, Kindly align them as we have done for other metrics. We don't indent after initial namespace of mlpack or mlpack::cv. You can refer to #2402 for reference.

Reply:
Done.

Reply:
Adding the tests in the next commit.

Reply:
Great, Thanks a lot.

Reply:
@kartikdutt18 On running the test, I am getting an std::bad_alloc exception. I am not really able to figure out why. Is it because there are so many(5) arma matrices?

Reply:
>  Is it because there are so many(5) arma matrices?

I don't think, do you mind pushing the tests as well. I'll take a closer look at the maths tomorrow evening.

Reply:
Done, what should I do in the meanwhile to get to know more about the repo as I am planning to devote some time to contributing to this repository. Can you also suggest me some relevant math and programming sources? Thanks in advance.

Reply:
I think you can try out some tutorial that are present in this repo. You could also clone and run some examples from mlpack/examples. You can also run these program on C++ notebooks at lab.mlpack.org. Other than this we also have some other good first issues if you want to take them up. For contributions you can also take a look at [this](https://www.mlpack.org/community.html).
For programming there are a ton of good websites. Have you tried codeforces or HackerRank. For Maths, personally I like the channel FlammableMaths.